### PR TITLE
V8 6.2.414.15

### PR DESCRIPTION
**Pull request by stesie**
###### Motivation for this change

Current version of V8 shipped with NixOS 5.4 is pretty old, actually it was published a year ago.
Since then updating was problematic, since upstream changed from Gyp to GN build system

This pull request packages GN (again, it was included once and then removed) and introduces a new package `v8_6_x` which packages current V8 version from Chromium/V8 beta release channel -- currently version 6.2.414.15

I also added a little `update.sh` shell script that updates the package recipe to latest V8 dependencies so future updates should be way easier

###### Caveats

The reason for me adding a new package and not simply upgrading "v8" was that due to re-packaging currently two features are lost, that the old package had
* `static` flag, to trigger static compilation & linking
* macOS support, can't test that as I don't have Mac hardware at hand

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/v8/6_x.nix [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+  buildPhase = ''
+    ninja -C out.gn/x64.release/
+  '';
+
+  enableParallelBuilding = true;
+
+  installPhase = ''
+    install -vD out.gn/x64.release/d8 "$out/bin/d8"
+    install -vD out.gn/x64.release/mksnapshot "$out/bin/mksnapshot"
```
Comment (Mic92) [inline pkgs/development/libraries/v8/6_x.nix]: Should this not be `${arch}.release`?

Comment (stesie) [reply pkgs/development/libraries/v8/6_x.nix]: Indeed it should, fixed

**Comment by Mic92** [comment]
It fails to build php-v8 in my case:

```
yes
checking whether to enable PDO support... yes
checking for mysql_commit in -lmysqlclient_r... checking for PDO_DBLIB support via FreeTDS... no
checking for Firebird support for PDO... no
checking for MySQL support for PDO... yes
checking for MySQL UNIX socket location... no
checking for mysql_config... /nix/store/3qmydxbzx7wvs759gslm744bqzp6zaq4-mariadb-client-10.1.25-dev/bin/mysql_config
checking for mysql_commit in -lmysqlclient_r... yes
checking for PDO includes... checking for PDO includes... /tmp/nix-build-php-7.0.22.drv-0/php-7.0.22/ext
checking Oracle OCI support for PDO... no
yes
checking for ODBC v3 support for PDO... no
checking for PostgreSQL support for PDO... yes
checking for pg_config... not found
configure: error: Cannot find libpq-fe.h. Please specify correct PostgreSQL installation path
checking for PDO includes... checking for PDO includes... /tmp/nix-build-php-7.1.8.drv-0/php-7.1.8/ext
checking Oracle OCI support for PDO... no
checking for ODBC v3 support for PDO... no
checking for PostgreSQL support for PDO... yes
checking for pg_config... not found
configure: error: Cannot find libpq-fe.h. Please specify correct PostgreSQL installation path
builder for ‘/nix/store/ab6yxiazghxp6l85hzv446n4dyi5pip8-php-7.0.22.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/w7ssz0dg86kaa510lv51d04g9k5ksahb-php-v8-0.1.9.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/fnkqls86z8d3y8p12jjnk4jafaw6wg2i-php-v8js-1.4.1.drv’: 1 dependencies couldn't be built
builder for ‘/nix/store/p6ni55z6kvl0k12zhwi97lqvm5s5z7cz-php-7.1.8.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/k6rm6y30dz6la2d2kxkrg464xy60v6rk-php-v8-0.1.9.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/93ps150nzvjw56rlhllykwincc4a1rjf-php-v8js-1.4.1.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/93ps150nzvjw56rlhllykwincc4a1rjf-php-v8js-1.4.1.drv’, ‘/nix/store/fnkqls86z8d3y8p12jjnk4jafaw6wg2i-php-v8js-1.4.1.drv’, ‘/nix/store/k6rm6y30dz6la2d2kxkrg464xy60v6rk-php-v8-0.1.9.drv’, ‘/nix/store/w7ssz0dg86kaa510lv51d04g9k5ksahb-php-v8-0.1.9.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
The invocation of "nix-shell --keep-going -j4 --option build-use-sandbox true -p php71Packages.v8 -p php70Packages.v8js -p php70Packages.v8 -p php71Packages.v8js -p gn -p v8_6_x" failed
$ git worktree prune
nix-review 29726  38.58s user 4.94s system 7% cpu 10:06.03 total
```

**Comment by stesie** [comment]
@Mic92 are you sure it's related to my change? I cannot reproduce that and it errors out building php-7.0.22 and php-7.1.8, which just are a dependency to php-v8.

**Comment by Mic92** [comment]
Probably not, but it happens on master. Looks pretty similar to breakage: https://github.com/NixOS/nixpkgs/pull/29602/commits/032c50d45b3fd1ebb4af592b83e776ed7004b4c0

**Comment by stesie** [comment]
Yes, it was broken by commit b0280f598e4b3f6ebf33ad2115734e4735df443a

... which splited pg-config into a separate, new build output "dev"

**Comment by copumpkin** [comment]
Looks like @abbradar tried to delete this a while back: https://github.com/NixOS/nixpkgs/commit/003fa8fed5480d3e22d518aeeba1d62809809279#diff-44657b1b205614922a0666e1fb8ca9b4

**Pull request merged**